client/trino-client/src/main/java/io/trino/client/ProtocolHeaders.java

```diff
@@ -0,0 +1,203 @@
+    }
+
+    public String requestCatalog()
+    {
+        return prefix + "Catalog";
+    }
+
+    public String requestSchema()
+    {
+        return prefix + "Schema";
+    }
+
+    public String requestPath()
+    {
+        return prefix + "Path";
+    }
+
+    public String requestTimeZone()
+    {
+        return prefix + "Time-Zone";
+    }
+
+    public String requestLanguage()
+    {
+        return prefix + "Language";
+    }
+
+    public String requestTraceToken()
+    {
+        return prefix + "Trace-Token";
+    }
+
+    public String requestSession()
+    {
+        return prefix + "Session";
+    }
+
+    public String requestRole()
+    {
+        return prefix + "Role";
+    }
+
+    public String requestPreparedStatement()
+    {
+        return prefix + "Prepared-Statement";
+    }
+
+    public String requestTransactionId()
+    {
+        return prefix + "Transaction-Id";
+    }
+
+    public String requestClientInfo()
+    {
+        return prefix + "Client-Info";
+    }
+
+    public String requestClientTags()
+    {
+        return prefix + "Client-Tags";
+    }
+
+    public String requestClientCapabilities()
+    {
+        return prefix + "Client-Capabilities";
+    }
+
+    public String requestResourceEstimate()
+    {
+        return prefix + "Resource-Estimate";
+    }
+
+    public String requestExtraCredential()
+    {
+        return prefix + "Extra-Credential";
+    }
+
+    public String responseSetCatalog()
+    {
+        return prefix + "Set-Catalog";
+    }
+
+    public String responseSetSchema()
+    {
+        return prefix + "Set-Schema";
+    }
+
+    public String responseSetPath()
+    {
+        return prefix + "Set-Path";
+    }
+
+    public String responseSetSession()
+    {
+        return prefix + "Set-Session";
+    }
+
+    public String responseClearSession()
+    {
+        return prefix + "Clear-Session";
+    }
+
+    public String responseSetRole()
+    {
+        return prefix + "Set-Role";
+    }
+
+    public String responseAddedPrepare()
+    {
+        return prefix + "Added-Prepare";
+    }
+
+    public String responseDeallocatedPrepare()
+    {
+        return prefix + "Deallocated-Prepare";
+    }
+
+    public String responseStartedTransactionId()
+    {
+        return prefix + "Started-Transaction-Id";
+    }
+
+    public String responseClearTransactionId()
+    {
+        return prefix + "Clear-Transaction-Id";
+    }
+
+    public static ProtocolHeaders detectProtocol(Optional<String> alternateHeaderName, Set<String> headerNames)
+            throws ProtocolDetectionException
+    {
+        requireNonNull(alternateHeaderName, "alternateHeaderName is null");
+        requireNonNull(headerNames, "headerNames is null");
+
+        if (alternateHeaderName.isPresent() && !alternateHeaderName.get().equalsIgnoreCase("Trino")) {
+            if (headerNames.contains("X-" + alternateHeaderName.get() + "-User")) {
+                if (headerNames.contains("X-Trino-User")) {
```

Comment:
This code assumes the user header (either `X-Trino-User` or `X-Alternate-User`) is always set, so it can use as the detection marker.
Thus, it should validate it is the case: if neither `X-Trino-User` or `X-Alternate-User` is set, it should throw `ProtocolDetectionException`